src/providers/mod.rs

```diff
@@ -330,30 +342,35 @@ mod tests {
             .take(maxlen * 2)
             .collect::<String>();
         // simple hostname
-        assert_eq!(try_write_hostname("hostname7"), "hostname7");
+        assert_eq!(try_write_hostname("hostname7", false), "hostname7");
         // simple FQDN
         assert_eq!(
-            try_write_hostname("hostname7.example.com"),
+            try_write_hostname("hostname7.example.com", false),
             "hostname7.example.com"
         );
+        // short form hostname
+        assert_eq!(
+            try_write_hostname("hostname7.example.com", true),
+            "hostname7"
+        );
```

Comment:
Let's at least also test short hostnames where the input value doesn't have an FQDN.